src/Api/PubnubApi/EndPoint/PubSub/UnsubscribeOperation.cs

```diff
@@ -54,7 +58,30 @@ public UnsubscribeOperation<T> QueryParam(Dictionary<string, object> customQuery
 
         public void Execute()
         {
-            Unsubscribe(subscribeChannelNames, subscribeChannelGroupNames);
+            if (config.EnableEventEngine && subscribeEventEngineFactory != null) {
+                if (subscribeEventEngineFactory.hasEventEngine(PubnubInstance.InstanceId)) {
+                    var subscribeEventEngine = subscribeEventEngineFactory.getEventEngine(PubnubInstance.InstanceId);
+                    subscribeEventEngine.Channels = subscribeEventEngine.Channels.Except(this.subscribeChannelNames).ToArray();
+                    subscribeEventEngine.Channelgroups = subscribeEventEngine.Channelgroups.Except(this.subscribeChannelGroupNames).ToArray();
+                    subscribeEventEngine.eventQueue.Enqueue(new SubscriptionChangedEvent() {
+                        Channels = subscribeEventEngine.Channels, ChannelGroups = subscribeEventEngine.Channelgroups
+                    });
+                    if (ChannelLocalUserState.TryGetValue(PubnubInstance.InstanceId, out var userState)) {
+                        foreach (var channel in this.subscribeChannelNames ?? Array.Empty<string>()) {
```

Comment:
This line doesn't compile for some versions of dotnet. 
```suggestion
                        foreach (var channel in this.subscribeChannelNames ?? new string[0]) {
```

Reply:
👍🏻